### PR TITLE
Fix: document.body is null error in Firefox

### LINKED ISSUE #2318: Firefox new tab Uncaught TypeError: document.body is null
## 💡 Summary

Firefox open in note or something in new tab with mid key got error Uncaught TypeError: document.body is null and stuck at loading screen, refresh page will load everything normally

## 🙂 Expected Behavior

load MiOS

## ☹️ Actual Behavior

stuck at loading screen

## 📌 Environment
v11.37.1-20221204012750

Follow fix not working:
the base.pug load js before body,
a dirty fix would be change line 363 in init.ts to "setTimeout(function(){os.init(() => {" and close it without time at line 514

### PR DESCRIPTION
**Pull request by mei23**
## Summary
Fix #2318
Wait for DOMContentLoaded before running script.

Originally script defer should work.
However, Firefox seems to ignore it in some cases.

### CODE REVIEW

**Comment by mei23** [comment]
なお、v12はv12.114.0あたりにあった (importだからちょっと違う)
https://github.com/misskey-dev/misskey/commit/44f560b453fe1161cc6e1dcad6e48fd0ac812cd4

**Comment by mei23** [comment]
v12はimportだからともかく、-v11はdefer付いてるのになぜ起きるのか不明。

File: src/client/app/boot.js [hunk 1]
```diff
@@ -6,166 +6,176 @@
...
-			cache: 'no-cache'
-		});
+		// Load an app script
+		// Note: 'async' make it possible to load the script asyncly.
+		//       'defer' make it possible to run the script when the dom loaded.
+		const script = document.createElement('script');
+		script.src = `/assets/${app}.${ver}.js`;
+		script.async = true;
+		script.defer = true;
+		head.appendChild(script);
```
Comment (mei23) [inline src/client/app/boot.js]: TODO: このあたりに局所化する

**Comment by mei23** [comment]
確かにFirefoxはたまにscript deferがあっても通過してしまう模様

**Comment by mei23** [comment]
```html
<html>
<head>
<script src=b.js async defer></script>
<script>
  for( let i = 0; i < 1000000000; i++)
  {}
</script>
</head>
<body></body>
</html>
```
みたいな呼び出しで、Firefoxで発生することと解決することを確認

**Comment by saschanaz** [comment]
https://html.spec.whatwg.org/multipage/scripting.html#attr-script-async

>For [classic scripts](https://html.spec.whatwg.org/multipage/webappapis.html#classic-script), if the [async](https://html.spec.whatwg.org/multipage/scripting.html#attr-script-async) attribute is present, then the classic script will be fetched [in parallel](https://html.spec.whatwg.org/multipage/infrastructure.html#in-parallel) to parsing and evaluated as soon as it is available (potentially before parsing completes). If the [async](https://html.spec.whatwg.org/multipage/scripting.html#attr-script-async) attribute is not present but the [defer](https://html.spec.whatwg.org/multipage/scripting.html#attr-script-defer) attribute is present, then the classic script will be fetched [in parallel](https://html.spec.whatwg.org/multipage/infrastructure.html#in-parallel) and evaluated when the page has finished parsing. If neither attribute is present, then the script is fetched and evaluated immediately, blocking parsing until these are both complete.

asyncとdeferが共にあるとasyncの方が優先されるそうです

**Comment by saschanaz** [comment]
（asyncを削除するべき）

**Comment by mei23** [comment]
たしかに defer だけにしたら解決した

**Pull request merged**